cogs/challenges.py

```diff
@@ -35,42 +35,65 @@ async def on_raw_reaction_add(self, payload):  # Participant role.
     async def on_message(self, message):  # Submitted role.
 
         if message.channel.id in (680851820587122700, 713841306253656064):  # weekly
-            
+
             if message.author.id == self.bot.user.id:
                 return
-            
+
             if message.author.bot:
                 return await message.delete()
-            
-            if message.channel.id == 680851820587122700:  # weekly 1 
+
+            if message.channel.id == 680851820587122700:  # weekly 1
                 submitted = self.bot.guild.get_role(687417501931536478)
                 submission_channel = self.bot.guild.get_channel(729453161885990924)
-            
+
             else:  # weekly 2
                 submitted = self.bot.guild.get_role(715676464573317220)
                 submission_channel = self.bot.guild.get_channel(729453201081761862)
 
             if submitted not in message.author.roles:
                 await message.delete()
                 attach = message.attachments and message.attachments[0]
-                
+
                 if not attach:
-                    msg = f"{message.author.mention} make sure to __upload a " \
-                          f"python file__ that only includes the code required " \
-                          f"for the challenge!"
+                    msg = (
+                        f"{message.author.mention} make sure to __upload a "
+                        f"file__ that only includes the code required "
+                        f"for the challenge!"
+                    )
                     return await message.channel.send(msg, delete_after=10.0)
-                
-                content = "```py\n" + (await attach.read()).decode("u8").replace("`", "\u200b`") + "```"
+
+                filetype = attach.filename.split(".")[-1]
+
+                if len(filetype) > 4 or len(filetype) == 0:  # Most filetypes are 2-3 chars, 4 just to be safe
+                    return await message.channel.send(
+                        "Please upload a file with a __non-empty file extension shorter than 5 chars__!",
+                        delete_after=10.0,
+                    )
+
+                content = (
+                    f"```{filetype}\n"
+                    + (await attach.read()).decode("u8").replace("`", "\u200b`")
+                    + "```"
+                )
                 if len(content) > 2040:
-                    msg = f"{message.author.mention} attachment can't be __more" \
-                          f" than 2040 characters__."
+                    msg = (
+                        f"{message.author.mention} attachment can't be __more"
+                        f" than 2040 characters__."
+                    )
                     return await message.channel.send(msg, delete_after=10.0)
```

Comment:
Actually the embed description length can be 4096 characters so we can allow longer codes.